RIOT-OS/RIOT
# boards/stm32gx: move kconfig clock configuration to cpu level

### Contribution description

This PR reworks a bit the Kconfig files used to configure the clock of STM32 G0 and G4 families.

There's also a small adaptation of the main Kconfig file in cpu/stm32 to only include the related Kconfig files based on CPU_FAM variable. This is cleaner IMHO since it doesn't mix information of unrelated CPU variants in menuconfig also allows to automatically include clock kconfig of a given family.




### Testing procedure

- A green Murdock
- Ensure `make menuconfig` still shows the clock options

<details><summary>on g0</summary>

![image](https://user-images.githubusercontent.com/1375137/97011228-b300cc00-1546-11eb-98e6-adfae2312660.png)

</details>

<details><summary>on g4</summary>

![image](https://user-images.githubusercontent.com/1375137/97011128-96649400-1546-11eb-8d71-81a2c815ce4a.png)

</details>




### Issues/PRs references

Tick another item in #14975
